numpy/_core/src/umath/loops_autovec.dispatch.c.src

```diff
@@ -61,6 +61,17 @@ NPY_NO_EXPORT void NPY_CPU_DISPATCH_CURFX(@TYPE@_reciprocal)
     UNARY_LOOP_FAST(@type@, @type@, *out = 1.0 / in);
 }
 
+/**begin repeat1
```

Comment:
Not a major worry, but just wondering: Aren't there loops for `np.positive` already? Is it possible to reuse those rather than define new ones?

Reply:
I think there's a loop per each dtype and function, like `@TYPE@_@kind@`.

Here, for `positive` we have:
```c
NPY_NO_EXPORT void NPY_CPU_DISPATCH_CURFX(@TYPE@_positive)
(char **args, npy_intp const *dimensions, npy_intp const *steps, void *NPY_UNUSED(func))
{
    UNARY_LOOP_FAST(@type@, @type@, *out = +in);
}
```
I think for `floor`, `ceil`, and `trunc` we need separate ones anyway to follow the convention `dtype_func`. Or do you mean changing `@TYPE@_positive` to `@TYPE@_@kind@` and adding an inner `repeat1` loop here? (with `#kind = floor, ceil, trunc, positive#`)

Reply:
yes, an alias seems the right solution, to avoid compiling the same code four times!

Reply:
Done! I added `#define` alias pointing to `positive` loops for `floor`, `ceil`, and `trunc` ufuncs.

But this alias covers only ints, as `positive` doesn't support boolean dtype. For boolean dtype I added separate no-op loops (I didn't find other ufunc that would be a no-op for booleans), is it OK?

Reply:
If `positive` doesn't support it, perhaps `ceil`, `trunc`, etc., should not support it either. My tendency would be to either add support for `positive` or drop support for the others for now.

Note that `+True` gives `1` and `-True` gives -1, so that is an argument against implementing `np.positive` as a copy for booleans. But perhaps also an argument against pretending to know what `ceil`, etc., should do.  So, somewhat prefer just punting on this, i.e., do integers only.

Reply:
Ok, it's still a question how we want `floor`/`ceil`/`trunc` to handle booleans (cast, no-op, or raise) and if we want to bind it to `positive` (and if `positive` should then support booleans).

Reply:
I also like the idea of adding support for integers only, and casting bools to ints:
```python
In [1]: np.ceil(np.array([1, 2, 3]).astype(bool))
Out[1]: array([1, 1, 1], dtype=int8)
```

Reply:
My preference would be just not to touch `bool` for now (and thus continue to raise), i.e., punt on the decision of how exactly to treat bools.

Reply:
Just for reference, for `np.negative` we also do not follow python, but have specific text (unlike for `np.positive`, which was added more recently),
```
In [2]: np.negative([True])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 np.negative([True])

TypeError: The numpy boolean negative, the `-` operator, is not supported, use the `~` operator or the logical_not function instead.

In [3]: np.positive([True])
---------------------------------------------------------------------------
UFuncTypeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 np.positive([True])

UFuncTypeError: ufunc 'positive' did not contain a loop with signature matching types <class 'numpy.dtypes.BoolDType'> -> None
```

Reply:
It would go to the first type that matches, which is either uint8 or int8 (dunno).

Reply:
It casts to `int8`:
```python
In [1]: np.ceil(np.array([1, 0, 1]).astype(bool))
Out[1]: array([1, 0, 1], dtype=int8)
```
Then let me update it here.

Reply:
Done! Ready from my side!